### PR TITLE
[Hub Generated] Review request for Microsoft.KeyVault to add version preview/2021-04-01-preview

### PR DESCRIPTION
**Pull request by jiacheng-L**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/jiacheng-L/azure-rest-api-specs/dev-keyvault-Microsoft.KeyVault-2021-04-01-preview-bugFix...master/).

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Review by JeffreyRichter — APPROVED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json [hunk 1]
```diff
@@ -691,72 +691,6 @@
         ]
       }
     },
-    "/subscriptions/{subscriptionId}/resources": {
```
Comment (heaths) [inline specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:694]: This breaks a lot of scripts, including ones we use to monitor cost analysis for the Azure SDK. Why is listing vaults in a sub being removed? Could you comment here, or even update the PR description with reasoning? This is a pretty big change.

**Review by heaths — CHANGES_REQUESTED** [review]

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json (hunk 1 above)
Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:694]: Hi @heaths , the actual key vault list API is on line 470: “/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.KeyVault/vaults”
The removing API seems doesn’t belong to us. When customer call GET with the removing url: "/subscriptions/{subscriptionId}/resources", it goes to another service endpoint, which I believed it is here: [Microsoft.Resource](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/resources/resource-manager/Microsoft.Resources/stable/2021-01-01/resources.json#L2881). So this make us unable to control this API as well as its version. I am not totally sure if I am wrong here and I don’t know why this API is listed in our code for a long time.

**Comment by zhenglaizhang** [comment]
@msyyc @ArcturusZhang pls help take a look at the python & go sdk breaking change

**Comment by jiacheng-L** [comment]
Synced with team offline, we don't need to modify this API. It is an ARM API and we need to keep this in our code to have better customer experience. I will close this PR soon.

File: specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json (hunk 1 above)
Comment (heaths) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:694]: It [was in 2019-09-01](https://github.com/Azure/azure-rest-api-specs/blob/5b1c5d06c84680cfc4c5a179c56ec8b1d9a0e295/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keyvault.json#L694) GA, so this is breaking. But is that desirable in this case? I worry about clients already relying on it.

Comment (jiacheng-L) [reply specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/keyvault.json:694]: It was in all the previous GA versions as well. But I am not sure if previous SDKs already have a way to due with it. I talked with team before. Looks like it is desirable to have this ARM API in our spec.

**Pull request closed without merging**